code/obj/critter/misc.dm

```diff
@@ -487,23 +489,54 @@
 			else
 				continue
 
+	attackby(obj/item/W as obj, mob/M as mob)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
+			if(prob(20))
+				src.visible_message("<span class='notice'>[src] chitters happily at the [W], it will no longer attack [M]!</span>")
+				friends += M
+				playsound(src.loc, "sound/misc/bugchitter.ogg", 50, 0)
+				M.drop_item()
+				qdel(W)
+				src.task = "thinking"
+				src.health = min(src.max_health, src.health + health_gain_from_food)
+				eat_twitch(src)
+			else
+				src.visible_message("<span class='notice'>[src] hated the [W]! It bit [M]'s hand!</span>")
+				take_bleeding_damage(M, M, rand(6,12), DAMAGE_STAB, 1)
+				playsound(src.loc, "sound/items/Wirecutter.ogg", 50, 0)
+				M.emote("scream")
+				M.drop_item()
+				qdel(W)
+				src.health = min(src.max_health, src.health + health_gain_from_food)
+				eat_twitch(src)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && (M in src.friends))
+			src.visible_message("<span class='notice'>[src] chitters happily at the [W].</span>")
+			playsound(src.loc, "sound/misc/bugchitter.ogg", 50, 0)
+			M.drop_item()
+			qdel(W)
+			src.health = min(src.max_health, src.health + health_gain_from_food)
+			eat_twitch(src)
+		..()
+
+
 	ChaseAttack(mob/M)
 		..()
-		if(prob(50))
+		if(prob(50) && (!ON_COOLDOWN(src, "scorpion_ability", 10 SECONDS)))
```

Comment:
It'd be cleaner if the cooldown check wasn't duplicated, maybe just move it before the if and make it:
```
if(ON_COOLDOWN(...))
  return
```